sklearn/pipeline.py

```diff
@@ -201,15 +201,12 @@ def _validate_steps(self):
         for t in transformers:
             if t is None or t == "passthrough":
                 continue
-            if not (hasattr(t, "fit") or hasattr(t, "fit_transform")) or not hasattr(
-                t, "transform"
-            ):
-                raise TypeError(
-                    "All intermediate steps should be "
-                    "transformers and implement fit and transform "
-                    "or be the string 'passthrough' "
-                    "'%s' (type %s) doesn't" % (t, type(t))
-                )
+            if (not (hasattr(t, "fit") or hasattr(t, "fit_transform")) or not
```

Comment:
This is overlapping with another PR at https://github.com/scikit-learn/scikit-learn/pull/22162, we can focus on that there.

Reply:
I didn't do it on purpose, I handled Git wrong but I put back the original code.